## By: Thenmani Sayebaba


# **Customer Segmentation for Targeted Mall Promotions**



In [22]:
pip install scikit-learn

# **When K = 3**

In [23]:
import pandas as pd

from sklearn.cluster import KMeans

data = pd.read_csv("Mall_Customers.csv")

X = data[['Annual Income (k$)', 'Spending Score (1-100)']]

k = 3  # change later

model = KMeans(n_clusters=k, random_state=42)

model.fit(X)

data['Cluster'] = model.labels_

summary = data.groupby('Cluster')[['Annual Income (k$)', 'Spending Score (1-100)']].agg(

    ['size', 'mean', 'std']

).reset_index()

print(f"\n=== Summary for k={k} ===")

print(summary)


=== Summary for k=3 ===
  Cluster Annual Income (k$)                       Spending Score (1-100)  \
                        size       mean        std                   size   
0       0                 38  87.000000  16.271348                     38   
1       1                 39  86.538462  16.312485                     39   
2       2                123  44.154472  16.037882                    123   

                         
        mean        std  
0  18.631579  10.915947  
1  82.128205   9.364489  
2  49.829268  19.694265  


# **Buisness Inference (when k= 3):**


# *   High Income, Low Spenders:
### Cluster 0 has rich 38 homogeneous members with mean income 87k with mean standard deviation 16.27 with mean spending score 18.63 and mean standard deviation 10.92

# *   High Income, High Spenders:
### Cluster 1 has rich 39 homogeneous customers with mean income 86.54k with mean standard deviation 16.31 with mean high spending score 82.13 with with less standard deviation 9.36

# *   Mid Income, Moderate Spenders:
### Cluster 2 has mid-income homogeneous customers with mean income 44.15k with mean standard deviation with mean spending score 49.83 but with widely dispersed spending pattern of mean standard deviation 19.69

# **Business must retain their best High Income, High spender homogeneous customers.**






# **When K = 4**

In [24]:
import pandas as pd

from sklearn.cluster import KMeans

data = pd.read_csv("Mall_Customers.csv")

X = data[['Annual Income (k$)', 'Spending Score (1-100)']]

k = 4  # Chosen cluster size

model = KMeans(n_clusters=k, random_state=42)

model.fit(X)

data['Cluster'] = model.labels_

summary = data.groupby('Cluster')[['Annual Income (k$)', 'Spending Score (1-100)']].agg(

    ['size', 'mean', 'std']

).reset_index()

print(f"\n=== Summary for k={k} ===")

print(summary)


=== Summary for k=4 ===
  Cluster Annual Income (k$)                       Spending Score (1-100)  \
                        size       mean        std                   size   
0       0                101  48.168317  14.519001                    101   
1       1                 39  86.538462  16.312485                     39   
2       2                 22  25.727273   7.566731                     22   
3       3                 38  87.000000  16.271348                     38   

                         
        mean        std  
0  43.396040  14.707875  
1  82.128205   9.364489  
2  79.363636  10.504174  
3  18.631579  10.915947  


#**Buisness Inference (when k= 4):**
###* **Homogeneity:**
###Yes, the standard deviations shrank within cluster when K=4. When K=3, the highest spending score deviation was 19.7 for cluster 2, Now when k=4, the high spending score deviation is reduced to 14.7 for cluster 0. So the customers are becoming homogeneous within clusters
###* **Size Balance:**
### Yes, Cluster 2 became very small with only 22 customers , when k=4. But they spend a lot with mean spending score 79.4. So it worth campaigning these small customer segments too.
## **Business can focus on campaigning homogenous Cluster 1 and 2 as they high spending score 82.13 and 79.63 with mean standrad deviations 9.4 and 10.5 respectively.**


# **When K = 6**

In [25]:
import pandas as pd

from sklearn.cluster import KMeans

data = pd.read_csv("Mall_Customers.csv")

X = data[['Annual Income (k$)', 'Spending Score (1-100)']]

k = 6  # Chosen cluster size

model = KMeans(n_clusters=k, random_state=42)

model.fit(X)

data['Cluster'] = model.labels_

summary = data.groupby('Cluster')[['Annual Income (k$)', 'Spending Score (1-100)']].agg(

    ['size', 'mean', 'std']

).reset_index()

print(f"\n=== Summary for k={k} ===")

print(summary)


=== Summary for k=6 ===
  Cluster Annual Income (k$)                       Spending Score (1-100)  \
                        size       mean        std                   size   
0       0                 77  56.051948   8.557693                     77   
1       1                 39  86.538462  16.312485                     39   
2       2                 22  25.727273   7.566731                     22   
3       3                 35  88.200000  16.399067                     35   
4       4                 12  24.583333   6.694072                     12   
5       5                 15  31.533333   9.538394                     15   

                         
        mean        std  
0  49.857143   6.500434  
1  82.128205   9.364489  
2  79.363636  10.504174  
3  17.114286   9.952154  
4   9.583333   5.282188  
5  35.866667   5.974549  


# **Buisness Inference (when k= 6):**
##* **Homogeneity:**
### Yes, the standard deviations shrank within cluster when K=6. Cluster 0, cluster 4, cluster 5 have preferrably low spending score deviations 6.5, 5.3, 5.97. The other clusters mean spending score deviation also fall between 9.4 to 10.5. So business can provide offer to cohesive, homogeneous customer segments.
##* **Size Balance:**
### Yes, the size of cluster 4 and 5 shrank so much with size 12 and 15 and with mean spedning score 9.5 and 35.9 respectively. Since every additional segment costs, business can skip campaigning these smaller clusters.
# **Business must focus on campaigning Cluster 0, 1, 2 and 3 and skip campaigning cluster 4 and 5 to save cost as they have few customers 12 and 15 respectively.**


## **Business Recommendations:**


## 1.   Yes, We can divide our loyalty membes into a handfule of distict customer segments, based on how much they earn and how they actually spend, so that each segment gets a promotion designed for it.
## 2.   We can have 4 customer segments (Cluster 0 with size 77, Cluster 1 with size 39, Cluster 2 with size 22, Cluster 3 with Size 35) and exclude smallaer clusters 4 and 5, as every additional segment costs campaigning cost.(Refer  k-means statistical anlaysis summary when k=6).



# **Budget Allocation**


Yes The $50,00 can be split proportionally by cluster size .

Cluster 0 receives 22,254 dollars (77 members),
Cluster 1 receives 11,272 dollars (39 members),
Cluster 2 receives 6,358 dollars (22 members),
Cluster 3 receives 10,116 dollars (35 members).

### * Cluster 0 (mid income, mid spending) → Will get business uplift because they are average shoppers — a small nudge helps a little
### * Cluster 1 (high income, high spending) → Will get business uplift by retaining or motivating these preferred customers.
### * Cluster 2 (low income, high spending) → Will get business uplift because they are already shopping enthusiastically — a discount makes them spend even more
### * Cluster 3 (high income, low spending) → Will get business upliftt because they have money but are not spending it — a good offer can change that

---
# **Feature Choice - Age Inclusion**

Yes, Age can be considered as another dimension. Business can measure the spending pattern of customers based on their gender and give promotion for the products the gender is frequently buying.

---
# **Measurement - Is segmentation really working?**

Within each segment, **randomly set aside 20% of members from receiving the promotion as control group and retain the remaining 80% as campaign group**. After a quarter, measure each of these group's spending pattern. For the campaign group detect the campaign cost from the revenue. Business can measure whether the segmented campaign actually worked by comparing the profit from 'Control Group' and 'Campaign Group'.
